File: eshop-integration/eAPI/v1/java/eAPI-v1-connector/src/main/java/cz/monetplus/mips/eapi/v1/connector/entity/CartItem.java
package cz.monetplus.mips.eapi.v1.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class CartItem extends ApiBase implements Signable {

	private static final long serialVersionUID = -3825192932302805075L;

	@XmlElement(nillable=false)
	public String name;

	@XmlElement(nillable=false)
	public int quantity;

	@XmlElement(nillable=false)
	public long amount;

	@XmlElement(nillable=true)
	public String description;

	public CartItem() {
	}

	public CartItem(String name, int quantity, long amount) {
		this.name = name;
		this.quantity = quantity;
		this.amount = amount;
	}

	public CartItem(String name, int quantity, long amount, String description) {
		this(name, quantity, amount);
		this.description = description;
	}

	@Override
	public String toSign() {
		StringBuffer sb = new StringBuffer(); 
		add(sb, name);
		add(sb, quantity);
		add(sb, amount);
		add(sb, description);
		deleteLast(sb);
		return sb.toString();
	}

}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/CartItem.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class CartItem : ApiBase
{
    public CartItem()
    {
    }

    public CartItem(string? name, int quantity, long amount)
    {
        Name = name;
        Quantity = quantity;
        Amount = amount;
    }

    public CartItem(string? name, int quantity, long amount, string? description)
    {
        Name = name;
        Quantity = quantity;
        Amount = amount;
        Description = description;
    }

    [JsonProperty("name")] public string? Name { get; set; }

    [JsonProperty("quantity")] public int Quantity { get; set; }

    [JsonProperty("amount")] public long Amount { get; set; }

    [JsonProperty("description")] public string? Description { get; set; }

    public string? ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Name);
        Add(sb, Quantity);
        Add(sb, Amount);
        Add(sb, Description);
        DeleteLast(sb);
        return sb.ToString();
    }
}